File: Leetcode 0102 Binary Tree Level Order Traversal [medium] [tree]/Binary Tree Level Order Traversal [bfs].java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
     
        List<List<Integer>> result = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        
        if (root == null) return result;
        
        queue.add(root);
        
        while (!queue.isEmpty()) {
            List<Integer> level = new ArrayList<>();
            int size = queue.size();
            
            for (int i = 0; i < size; i++) {
                TreeNode head = queue.poll();
                level.add(head.val);
                
                if (head.left != null) {
                    queue.add(head.left);
                }
                
                if (head.right != null) {
                    queue.add(head.right);
                }
            }
            
            result.add(level);
        }
        
        return result;
        
    }
}

File: Leetcode 0102 Binary Tree Level Order Traversal [medium] [tree]/Binary Tree Level Order Traversal [bfs].js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let result = [];
    let queue = [];
    
    if (root === null) {
        return result;
    }
    
    queue.push(root);
    
    while (queue.length !== 0) {
        let tempResult = [];
        let tempQue = [];
        
        while (queue.length !== 0) {
            let cur = queue.shift();
            
            if (cur.left) {
                tempQue.push(cur.left);
            }
            
            if (cur.right) {
                tempQue.push(cur.right);
            }
            
            tempResult.push(cur.val);
        }
        
        result.push(tempResult);
        queue = tempQue;
    }
    
    return result;
};